### PR TITLE
Test/add missing tests

### PR DESCRIPTION
**Pull request by petergoes**

### CODE REVIEW

File: lib/mapbox/add-layer.js [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
+  const paint = get(`paint[${styleId}]`, layer) || {}
 
-    const styledLayer = {
-      ...layer,
-      paint: { ...defaultPaint, ...paint },
-      layout: { ...defaultLayout, ...layout },
-    }
-
-    mapbox.addLayer(styledLayer)
+  const styledLayer = {
```
Comment (CindyvdVries) [inline lib/mapbox/add-layer.js]: Maybe for the consistency also use lodash merge here?

File: test/unit/layouts/default.spec.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      map.getters.datasetsInActiveTheme(),
+    )
+    expect(wrapper.vm.mapboxOptions).toEqual({
+      sources: map.getters.activeDatasetsLocations(),
+      style: preferences.modules.theme.state.active,
+    })
+  })
+
+  test('dispatches loadPointDataForLocation on load-locations mapbox event', () => {
+    const wrapper = shallowMount(Default, {
```
Comment (CindyvdVries) [inline test/unit/layouts/default.spec.js]: Test this one also with multiple locationsIds --> that it always uses only the first locationId to get the pointdata. 
Maybe even with multiple datasetIds.

File: test/unit/layouts/default.spec.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    })
+
+    wrapper.find('select').element.value = 'dark'
+    wrapper.find('select').trigger('change')
+
+    expect(preferences.modules.theme.mutations.setActive.mock.calls[0][1]).toBe(
+      'dark',
+    )
+  })
+})
```
Comment (CindyvdVries) [inline test/unit/layouts/default.spec.js]: Missing test for event: toggle-location-dataset (calling toggleLocationDataset)  from DataSetControlMenu component.

File: test/unit/layouts/default.spec.js (hunk 2 above)
Comment (petergoes) [reply test/unit/layouts/default.spec.js]: I only want to test things like how they are intended. When we actually implement support for multiple locationIds, then I want to add a test. The current scenario is sure to be out of date. Writing a test for a soon to be out of date scenario seems overhead

File: test/unit/layouts/default.spec.js (hunk 3 above)
Comment (petergoes) [reply test/unit/layouts/default.spec.js]: this is covered in the test `updates url when datasetId is switched on/off` right?

**Pull request merged**